reason_4.0/lib/core/minisite_templates/modules/events.php

```diff
@@ -3719,6 +3770,62 @@ function get_ical($events)
   		return $calendar -> get_icalendar_events();
 	}
 	/**
+	 * Get a json representation of a set of events
+	 * @param array $events entities
+	 * @return string JSON
+	 */
+	function get_json($events)
+	{
+		if(!is_array($events))
+		{
+			trigger_error('get_json needs an array of event entities');
+			return '';
+		}
+
+		$nav = $this->get_page_nav();
+		$page_url = $nav->get_full_url( $this->page_id, true, true );
+
+		$events = array_values(array_map(function($e) use ($page_url) {
+			// ensure that all events have an associated URL
+			if(!$e->get_value('url')) {
+				$e->set_value('url', $page_url.'?event_id='.$e->id());
+			}
+			$values = $e->get_values();
+
+			$values['calendar_record'] = $values['calendar_record'] ? (int)$values['calendar_record'] : null;
+			$values['dates'] = explode(", ", $values['dates']);
+			$values['datetime'] = $values['datetime'] ?: null;
+			$values['end_date'] = $values['end_date'] ?: null;
+			$values['frequency'] = $values['frequency'] ? (int)$values['frequency'] : null;
+			$values['friday'] = $values['friday'] == 'true';
+			$values['hours'] = $values['hours'] ? (int)$values['hours'] : null;
+			$values['id'] = (int)$values['id'];
+			$values['last_occurence'] = $values['last_occurence'] ?: null;
+			$values['latitude'] = $values['latitude'] ? (double)$values['latitude'] : null;
+			$values['longitude'] = $values['longitude'] ? (double)$values['longitude'] : null;
+			$values['minutes'] = $values['minutes'] ? (int)$values['minutes'] : null;
+			$values['monday'] = $values['monday'] == 'true';
+			$values['month_day_of_week'] = $values['month_day_of_week'] ?: null;
+			$values['monthly_repeat'] = $values['monthly_repeat'] ?: null;
+			// $values['new'] = $values['new'] == '1';
+			// $values['no_share'] = $values['no_share'] == '1';
+			$values['recurrence'] = $values['recurrence'] ?: null;
+			$values['registration'] = $values['registration'] ?: null;
```

Comment:
According to some quick research, the "elvis operator" (a condensed ternary expression that's short for `a ? a : b`) is supported on PHP5.3+. I'm not sure what Reason itself supports, but I can expand them to the longhand form if I need to.